lib/query.js

```diff
@@ -4575,6 +4575,15 @@ Query.prototype._castUpdate = function _castUpdate(obj, overwrite) {
     upsert = this.options.upsert;
   }
 
+  let strict;
+  if ('strict' in this._mongooseOptions) {
+    strict = this._mongooseOptions.strict;
+  } else if (schema && schema.options) {
```

Comment:
It looks like this logic breaks our test for #8961. I don't think this PR as written will work, because this code only runs on the top-level schema. So this will always force nested schemas to have the same `strict` mode as the top level schema.

I think the fix for this issue is going to end up in `lib/helpers/query/castUpdate.js` similar to the fix for #8961: https://github.com/Automattic/mongoose/commit/dd122fe9b17641a89244d93330a9dc460911c34b

Reply:
Got it, I think ok I understand now.

We were using Mongoose 5.x and we relied on the "feature/bug" that the top level schema settings would apply to the sub document / nested schemas since it only read the top-level schema.

Now that this is working properly, we just need to explicitly set `{ strict: false }` on all our nested schemas.

I will close out this PR thank you for clarifying!